### PR TITLE
feat: added guardrails

### PR DESCRIPTION
**Pull request by smyja**

### CODE REVIEW

File: aide/backend/backend_corcel.py [hunk 1]
```diff
@@ -1,37 +1,46 @@
...
+        instruction (str): The instruction to be passed to the LLM API
+        msg_history (list[dict]): The message history to be passed to the LLM API
+        **kwargs: Any additional arguments to be passed to the LLM API
+
+    Returns:
+        str: The output of the LLM API
+    """
+  
+    # Setup Corcel API client
+    api_key = "b7a139c3-8a8e-4d19-a754-27724054dd2f"
```
Comment (crazydevlegend) [inline aide/backend/backend_corcel.py]: putting API key in the code directly is not a good practice

File: aide/backend/test.py
Comment (crazydevlegend) [inline aide/backend/test.py:1]: lemme run this in a few hours.
by the way, can you please work inside the whole project - as written in the readme, we should be able to run using `aide data_dir="" goal="" evaluation=""` command

**Review by crazydevlegend — CHANGES_REQUESTED** [review]
overall, lgtm, while you still need some modifications

File: aide/backend/backend_corcel.py [hunk 2]
```diff
@@ -41,34 +50,31 @@ def query(
...
+                # Check if the output content is a valid JSON string
+                try:
+                    json.loads(output_content)
+                    return output_content
+                except json.JSONDecodeError:
+                    # If the output content is not a valid JSON string, construct a default response
+                    return output_content
+
+    # Handle errors or unsuccessful responses
+    print(f"Corcel API request failed with status code {response.status_code}")
```
Comment (crazydevlegend) [inline aide/backend/backend_corcel.py]: better to use `logger` than using print (as in other files)

File: aide/backend/backend_corcel.py [hunk 3]
```diff
@@ -41,34 +50,31 @@ def query(
...
-                info = {"finish_reason": finish_reason, **payload}
-    else:
-        # Handle errors or unsuccessful responses
-        print(f"Corcel API request failed with status code {response.status_code}")
-
-    # Return the processed output along with additional information
-    return output_content, req_time, None, None, info
+            
+                # Check if the output content is a valid JSON string
+                try:
```
Comment (crazydevlegend) [inline aide/backend/backend_corcel.py:68]: check this try-catch block, it returns the same result in both cases?

File: aide/backend/backend_corcel.py (hunk 2 above)
Comment (smyja) [reply aide/backend/backend_corcel.py]: okay

File: aide/backend/backend_corcel.py (hunk 3 above)
Comment (smyja) [reply aide/backend/backend_corcel.py:68]: resolved

**Pull request closed without merging**